pkg/kubelet/secret/secret_manager.go

```diff
@@ -169,7 +169,14 @@ func (s *secretStore) Get(namespace, name string) (*v1.Secret, error) {
 	data.Lock()
 	defer data.Unlock()
 	if data.err != nil || !s.clock.Now().Before(data.lastUpdateTime.Add(s.ttl)) {
-		secret, err := s.kubeClient.Core().Secrets(namespace).Get(name, metav1.GetOptions{})
+		opts := metav1.GetOptions{}
+		if data.secret != nil && data.err == nil {
+			// This is just a periodic refresh of a secret we successfully fetched previously.
+			// In this case, we just try to get data from apiserver cache to reduce the load
+			// on etcd (the cache is eventuallly consistent).
+			opts.ResourceVersion = "0"
```

Comment:
Also to prove my words from above, I'm seeing a bunch of logs like this in our kubemark runs:
```
I0125 13:47:20.199420       7 wrap.go:75] GET /api/v1/namespaces/e2e-tests-density-30-32-lxszp/secrets/default-token-9zsd0: (2.716521925s) 200 [[kubemark/v0.0.0 (linux/amd64) kubernetes/$Format] 104.197.210.74:41632]
```

2.7s for processing a simple GET operation is quite a lot...